src/snowflake/cli/_plugins/workspace/commands.py

```diff
@@ -15,22 +15,53 @@
 from __future__ import annotations
 
 import typer
+import yaml
 from snowflake.cli._plugins.nativeapp.artifacts import BundleMap
+from snowflake.cli._plugins.snowpark.commands import migrate_v1_snowpark_to_v2
+from snowflake.cli._plugins.streamlit.commands import migrate_v1_streamlit_to_v2
 from snowflake.cli._plugins.workspace.manager import WorkspaceManager
 from snowflake.cli.api.cli_global_context import get_cli_context
 from snowflake.cli.api.commands.decorators import with_project_definition
 from snowflake.cli.api.commands.snow_typer import SnowTyper
 from snowflake.cli.api.entities.common import EntityActions
 from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.api.secure_path import SecurePath
 
 ws = SnowTyper(
     name="ws",
-    hidden=True,
     help="Deploy and interact with snowflake.yml-based entities.",
 )
 
 
-@ws.command(requires_connection=True)
+@ws.command()
+def migrate(
+    **options,
+):
+    """Migrates the Snowpark and Streamlit project definition files form V1 to V2."""
+    pd = DefinitionManager().unrendered_project_definition
```

Comment:
I understand that we are trying to preserve env variables from being expanded, but that's not the only rendering we have. You could refer to variables like `<% ctx.snowpark.project_name %>` anywhere in the project definition file. This will not work anymore in v2 because it's mapped to a different location.

We should at least warn about these templates, because they won't work anymore in v2.

Reply:
I am not sure. Conversion of these is not straightforward. You would might need a map of their previous location to their new location in pdf v2... it's not something that can quickly be done.

Reply:
If you want to get this out quickly, I suggest:
1) either error out on templating or if templating is used outside of referencing ctx.env stuff...
2) or expand the template first, and provide the expanded template as pdf v2

Reply:
I went for option 1. Attempt to migrate templated `snowflake.yml` causes error. User has the option to manually override this. In that case, we will log warning, stating that there is no guarantee the migrated file will work.